cdap-ui/graphql/schema/programRecordSchema.graphql

```diff
@@ -0,0 +1,111 @@
+"""
+ Copyright © 2019 Cask Data, Inc.
+ 
+ Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ use this file except in compliance with the License. You may obtain a copy of
+ the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ License for the specific language governing permissions and limitations under
+ the License.
+"""
+
+interface ProgramRecord {
+
+  type: String!
+
+  app: String!
+
+  name: String!
+
+  description: String!
+}
+
+type MapReduce implements ProgramRecord {
+
+  type: String!
+
+  app: String!
+
+  name: String!
+
+  description: String!
+}
+
+type Spark implements ProgramRecord {
+
+  type: String!
+
+  app: String!
+
+  name: String!
+
+  description: String!
+}
+
+type Workflow implements ProgramRecord {
+
+  type: String!
+
+  app: String!
+
+  name: String!
+
+  description: String!
+
+  runs: [RunRecord]!
+
+  # Field added for composition
+  schedules: [ScheduleDetail]!
+}
+
+type RunRecord {
+
+  runid: String!
+
+  starting: String!
+
+  start: String!
+
+  end: String!
+
+  status: String!
+
+  # properties
+
+  # ProgramRunCluster cluster;
```

Comment:
nit Do we need this?

Reply:
These are fields that can be added to these types, since they map to a Java class in the back, but they do not have resolvers implemented. There is no issue with deleting them as long as there is a way for the schema owner to know what other fields could be fetch for these types.

Reply:
The schema resolver should only be looking at the REST api response right? Why would they be seeing the java class here?

Reply:
True. The schema resolvers could see the HTTP response and what other fields are available. Then, the schema can be updated if it is necessary.